dialects/matrix/matrix.thorin

```diff
@@ -146,7 +146,7 @@
 ///
 /// Follow the principle `ij <- ik,kj` (`out[i,j] = sum_k in1[i,k] * in2[k,j]`) by using mulplication as combination function and addition as reduction function.
 .lam .extern internal_mapRed_matrix_prod
-    ![m: .Nat, k: .Nat, l: .Nat, [p: .Nat, e:.Nat]] ->
+    ![m: .Nat, k: .Nat, l: .Nat, [p: .Nat, e:.Nat]]:
```

Comment:
In general, it seems more readable / I am more used to `: return`.
However, in this case, it seems a bit odd (well the large return type itself is probably odd).

I think it is an improvement overall.